tests/specs/events.js

```diff
@@ -0,0 +1,1010 @@
+    t.is(removeEvents[0].object.id, 'one', 'event passes object')
+  })
+})
+
+test('store.one("remove") with removing two', function (t) {
+  t.plan(2)
+
+  var db = dbFactory()
+  var store = db.hoodieApi()
+  var oneRemoveEvent = []
+  var removeEvents = []
+
+  store.one('remove', addEventToArray.bind(null, oneRemoveEvent))
+  store.on('remove', addEventToArray.bind(null, removeEvents))
+
+  store.add([
+    {id: 'one'},
+    {id: 'two'}
+  ])
+
+  .then(function () {
+    return store.remove(['one', 'two'])
+  })
+
+  .then(waitFor(function () {
+    return removeEvents.length
+  }, 2))
+
+  .then(function () {
+    t.is(oneRemoveEvent.length, 1, 'triggers 1 remove event')
+    t.is(oneRemoveEvent[0].object.id, 'one', 'event passes object')
+  })
+})
+
+test('store.one("remove") with remove all', function (t) {
+  t.plan(2)
+
+  var db = dbFactory()
+  var store = db.hoodieApi()
+  var oneRemoveEvent = []
+  var removeEvents = []
+
+  store.one('remove', addEventToArray.bind(null, oneRemoveEvent))
+  store.on('remove', addEventToArray.bind(null, removeEvents))
+
+  store.add([
+    {id: 'one'},
+    {id: 'two'}
+  ])
+
+  .then(function () {
+    return store.removeAll()
+  })
+
+  .then(waitFor(function () {
+    return removeEvents.length
+  }, 2))
+
+  .then(function () {
+    t.is(oneRemoveEvent.length, 1, 'triggers 1 remove event')
+    t.is(oneRemoveEvent[0].object.id, 'one', 'event passes object')
+  })
+})
+
+test('store.one("change") with adding one', function (t) {
+  t.plan(3)
+
+  var db = dbFactory()
+  var store = db.hoodieApi()
+  var changeEvents = []
+
+  store.one('change', addEventToArray.bind(null, changeEvents))
+
+  store.add({
+    foo: 'bar'
+  })
+
+  .then(waitFor(function () {
+    return changeEvents.length
+  }, 1))
+
+  .then(function () {
+    t.is(changeEvents.length, 1, 'triggers 1 change event')
+    t.is(changeEvents[0].eventName, 'add', 'passes the event name')
+    t.is(changeEvents[0].object.foo, 'bar', 'event passes object')
+  })
+})
+
+test('store.one("change") with adding two', function (t) {
+  t.plan(3)
+
+  var db = dbFactory()
+  var store = db.hoodieApi()
+  var oneChangeEvent = []
+  var changeEvents = []
+
+  store.one('change', addEventToArray.bind(null, oneChangeEvent))
+  store.on('change', addEventToArray.bind(null, changeEvents))
+
+  store.add([
+    {foo: 'bar'},
+    {foo: 'baz'}
+  ])
+
+  .then(waitFor(function () {
+    return changeEvents.length
+  }, 2))
+
+  .then(function () {
+    t.is(oneChangeEvent.length, 1, 'triggers 1 change event')
+    t.is(oneChangeEvent[0].eventName, 'add', 'passes the event name')
+    t.is(oneChangeEvent[0].object.foo, 'bar', 'event passes object')
+  })
+})
+
+test('store.on returns store', function (t) {
+  t.plan(1)
+
+  function noop () {
+  }
+
+  var db = dbFactory()
+  var store = db.hoodieApi()
+
+  t.ok(store.on('add', noop) &&
+       typeof store.on('add', noop).on === 'function', 'allows chaining')
```

Comment:
sorry for the nitpicking, but could you do 

```js
function noop () {}
```

in just one line, and add a variable

```
var isFunction = store.on('add', noop) && typeof store.on('add', noop).on === 'function'
t.ok(isFunction, 'allows chaining')
```

I think that's more readable? I can also do it if you're busy. Because the rest looks goldan :ok_hand: